src/celeritas/field/CartMapFieldParams.hh

```diff
@@ -0,0 +1,54 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/field/CartMapFieldParams.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Types.hh"
+#include "corecel/data/ParamsDataInterface.hh"
+
+#include "CartMapFieldData.hh"
```

Comment:
Is it possible to forward declare the `CartMapFieldParamsData` and use a PIMPL class to hide the data? This include ends up propagating Covfie downstream, and requires a bunch of `#if CELERITAS_USE_COVFIE`, which is antithetical to how we handle dependencies.

Reply:
I guess it would be possible, but other classes ( the propagator factory, the `CartMapField`) depends on ParamsData too, and would also propagate the covfie dependency. If we want to hide it, then `CartMapFieldParamsData` would have to use PIMPL